design-proposals/network-binding-plugin/network-binding-plugin.md

```diff
@@ -1189,6 +1196,8 @@ spec:
         passt:
           networkAttachmentDefinition: default/netbindingpasst
           sidecarImage: quay.io/kubevirt/network-passt-binding
+          resourceOverhead:
+            memory: 800Mi
```

Comment:
I think you will have to explain why not:
- Directly `memoryOverhead: 500Mi` (BTW, it should be 500 and not 800).
- With explicit `request` and being open for adding `limit`.

Reply:
runtimeClasses have a pretty similar concept, maybe worth aligning the API idea
https://kubernetes.io/docs/concepts/scheduling-eviction/pod-overhead/

ie.

    overhead:
      podFixed:
        memory: "120Mi"
        cpu: "250m"

Reply:
> runtimeClasses have a pretty similar concept, maybe worth aligning the API idea

Interesting, thanks for sharing this.
I think the API could be in sync with the general [resources](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#example-1) concept.
This allows to control the resource additions to the container, with the ability to extend it in the future to any type of resource and define both request and limits.

The oddity here, as I see it, is how we express the resources of a different container (i.e. compute) and still not lock the ability to do the same for the sidecar itself.

Reply:
@EdDev I'm late to review this, but I don't understand why this API needs to add anything to the compute container and not simply advertise the resources needed to run this binding container?!
Same way as it's done for hotplug / other containers?
https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/services/renderresources.go#L611

Reply:
> runtimeClasses have a pretty similar concept, maybe worth aligning the API idea https://kubernetes.io/docs/concepts/scheduling-eviction/pod-overhead/
> 
> ie.
> 
> ```
> overhead:
>   podFixed:
>     memory: "120Mi"
>     cpu: "250m"
> ```

@fabiand I doubt it aligns well. This overhead API was invented for kata and the overhead is being added to any pod created with this runtimeclass, I don't think it aligns.

Reply:
> @EdDev I'm late to review this, but I don't understand why this API needs to add anything to the compute container and not simply advertise the resources needed to run this binding container?! Same way as it's done for hotplug / other containers? https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/services/renderresources.go#L611

@EdDev do we expect all network binding to affect the compute container or is it just passt?
passt specifically uses a feature in qemu there for the `GetMemoryOverhead` functional can identify that passt will be used and add a passt specific overhead - this is instead of a kubevirt CR API.
What do you think?

Reply:
@vladikr my comment was only about the API design, not about using the POD api.

Reply:
@vladikr Hi
I see that according to the network binding plugin design document, one of the [integration points](https://github.com/kubevirt/community/blob/main/design-proposals/network-binding-plugin/network-binding-plugin.md#integration-points) that seems legit is the [pod definition phase](https://github.com/kubevirt/community/blob/main/design-proposals/network-binding-plugin/network-binding-plugin.md#pod-definition--creation)
There were no exceptions regarding resource rendering, and Kubevirt CR was advised as a potential API to extend in order to integrate into that point.

Given that the network binding design was accepted, and also the implementation of this design also got into the codebase, I am not sure how to proceed, I'm in favor of accepting this design. 

Can you please advice @vladikr ?